client/src/components/chats/Chats.js

```diff
@@ -1,5 +1,85 @@
+import { useRef } from 'react' // add useState
+import { useSelector } from 'react-redux' // add useDispatch
+import { sortArrayOfObjects } from '../../utils/helper'
+import StartAChatButton from '../button/StartAChatButton'
+import ModalStartAChat from '../modal/ModalStartAChat'
+import LatestChats from './LatestChats'
+
+// temporary mock data that will be replaced with data from chats store
+import { chatsMockData } from './mock_messages'
+
 function Chats() {
-  return <div>Chats Component</div>
+  // ******todo******
+  // import action from chat store and dispatch an action that sets the current
+  // friend id with an id of a friend that has been clicked
+  // ****************
+
+  // Create the state variable for the selected friend to start a chat
+  // const [selectedFriendId, setSelectededFriendId] = useState()
+
+  // Get the reducer dispatch action
+  // const dispatch = useDispatch()
+
+  // Create handler for selecting the friend when clicking either last message or a friend
+  // const handleSelectFriend = (selectededFriendId) => {
+  //   setSelectedFriendId(selectededFriendId) - set the selected friend id
+  //   dispatch(currentFriend(selectedFriendId) - dispatch an action
+  //   modalRef.current.open && onCloseModal() - close the modal if opened
+  // }
+
+  // Get friends list from the user state slice
+  const friends = useSelector(({ user }) => user.friends)
+  // Get the modal ref
+  const modalRef = useRef()
+  // Get the list of the latest messages, one message per contact
+  const latestChats = chatsMockData.reduce(
```

Comment:
So after sending a request to the messages service, the received data is set in the store like so:
```
import { getMessages } from '../../services/messages'
import { setMessages } from '../../store/features/messages/messagesSlice'

useEffect(() => {
  chatsData = await getMessages(id, token)
  dispatch(setMessages(chatsData))
}, [])
```

Ideally, the component should re-render each time a new message is sent/received to update the data. We can achieve that by passing the messages state - imported from the messages store via useSelector - to the useEffect array argument. But it's fine for now to update store on first render only.

Reply:
Thank you for that comment, I'll apply the requested changes.